packages/@aws-cdk/aws-apigatewayv2-integrations/lib/http/aws.ts

```diff
@@ -0,0 +1,167 @@
+import * as iam from '@aws-cdk/aws-iam';
+import { IStateMachine } from '@aws-cdk/aws-stepfunctions';
+import { Construct } from 'constructs';
+import { AwsServiceIntegration, AwsServiceIntegrationProps } from './private/integration';
+
+/**
+ * The common Step Functions integration resource for HTTP API
+ */
+abstract class StepFunctionsIntegration extends AwsServiceIntegration {
+
+  /**
+   *
+   * @internal
+   */
+  protected _integrationService(): string {
+    return 'StepFunctions';
+  }
+}
+
+/**
+ * Step Functions StartExecution integration properties
+ */
+export interface StepFunctionsStartExecutionIntegrationProps extends AwsServiceIntegrationProps {
+
+  /**
+   * The state machine to be executed
+   */
+  readonly stateMachine: IStateMachine;
+
+  /**
+   * The execution name
+   *
+   * @default - undefined
+   */
+  readonly name?: string;
+
+  /**
+   * The input parameters of execution
+   *
+   * @default - undefined
+   */
+  readonly input?: any;
+
+  /**
+   * The region of state machine
+   *
+   * @default - undefined
+   */
+  readonly region?: string;
+}
+
+/**
+ * The StepFunctions-StartExecution integration resource for HTTP API
+ */
+export class StepFunctionsStartExecutionIntegration extends StepFunctionsIntegration {
+
+  constructor(private readonly _scope: Construct, private readonly _props: StepFunctionsStartExecutionIntegrationProps) {
+    super(_props);
+  }
+
+  /**
+   *
+   * @internal
+   */
+  protected _integrationAction(): string {
+    return 'StartExecution';
+  }
+
+  /**
+   *
+   * @internal
+   */
+  protected _fulfillRole(credentialsRole: iam.IRole): void {
+    this._props.stateMachine.grantStartExecution(credentialsRole);
+    credentialsRole.attachInlinePolicy(
+      new iam.Policy(this._scope, 'AllowSfnSyncExec', {
+        statements: [
+          new iam.PolicyStatement({
+            actions: ['states:StartSyncExecution'],
+            effect: iam.Effect.ALLOW,
+            resources: ['*'],
+          }),
+        ],
+      }),
+    );
```

Comment:
I believe this should be removed.

Reply:
This attaches the policy and fixes the permission bug mentioned above.

Reply:
but this is for StartExecution, not StartSyncExecution right? They are different constructs and therefore requires different permissions.

Reply:
yes I have fixed that in the push